lib/_stream_writable.js

```diff
@@ -515,21 +521,35 @@ function needFinish(state) {
 }
 
 function prefinish(stream, state) {
-  if (!state.prefinished) {
-    state.prefinished = true;
-    stream.emit('prefinish');
+  if (!state.prefinished && !state.finalCalled) {
+    if (typeof stream._final === 'function') {
+      state.pendingcb++;
+      state.finalCalled = true;
+      process.nextTick(() => {
```

Comment:
are you also talking about `stream._final((err) => {` one because I'm not sure we can extract that one